Python/Product/PythonTools/PythonTools/CodeCoverage/CoveragePyConverter.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.XPath;
+
+namespace Microsoft.PythonTools.CodeCoverage {
+    /// <summary>
+    /// Loads code coverage data from the XML format saved by coverage.py.
+    /// </summary>
+    class CoveragePyConverter {
+        private readonly Stream _input;
+        private readonly string _coverageXmlBasePath;
+
+        public CoveragePyConverter(string baseDir, Stream input) {
+            _coverageXmlBasePath = baseDir;
+            _input = input;
+        }
+
+        public CoverageFileInfo[] Parse() {
+            XmlDocument doc = new XmlDocument();
+            doc.Load(_input);
+            Dictionary<string, HashSet<int>> data = new Dictionary<string, HashSet<int>>();
+
+            var root = doc.DocumentElement.CreateNavigator();
+            string basePath = "";
+            foreach (XPathNavigator source in root.Select("/coverage/sources/source")) {
```

Comment:
What happens if this path is not found? Do we crash or just not return any results?